src/keys/public.rs

```diff
@@ -4,14 +4,49 @@
 //
 // Copyright (c) DUSK NETWORK. All rights reserved.
 
-use super::secret::NoteSecretKey;
-use dusk_bytes::{Error, HexDebug, Serializable};
-use dusk_jubjub::{JubJubAffine, JubJubExtended, GENERATOR_EXTENDED};
+//! # Note Public Key Module
+//!
+//! This module provides the public key components for the Schnorr signature
+//! scheme, necessary for vverifying note validity. It includes single and
+//! pair-based public keys. Public keys in this context are points on the JubJub
+//! elliptic curve generated from the [`NoteSecretKey`], which provide the basis
+//! for signature verification.
+use dusk_bytes::{DeserializableSlice, Error, HexDebug, Serializable};
+use dusk_jubjub::{
+    JubJubAffine, JubJubExtended, GENERATOR_EXTENDED, GENERATOR_NUMS_EXTENDED,
+};
+
+use crate::NoteSecretKey;
 
 #[cfg(feature = "rkyv-impl")]
 use rkyv::{Archive, Deserialize, Serialize};
 
-/// Structure repesenting a [`NotePublicKey`]
+/// Structure repesenting a [`NotePublicKey`] an extended point on the JubJub
+/// curve [`JubJubExtended`]. This public key allows for the verification of
+/// signatures created with its corresponding secret key without revealing the
+/// secret key itself.
+///
+/// ## Fields
+///
+/// - `0`: The [`JubJubExtended`] point representing the public key on the
+///   JubJub curve. This field is not directly accessible due to being private.
```

Comment:
I'm not sure if 'Fields' make so much sense here. I find the `0` very confusing.
In my opinion it's enough to say what you wrote above, that the `NotePublicKey` is a `JubJubExtended`